src/Http/Client/FormDataPart.php

```diff
@@ -198,10 +217,16 @@ public function __toString()
         if ($this->_disposition) {
             $out .= 'Content-Disposition: ' . $this->_disposition;
             if ($this->_name) {
-                $out .= '; name="' . $this->_name . '"';
+                $out .= '; name="' . preg_replace('/[^\x20-\x7E]/', '', $this->_name) . '"';
+                if ($this->_charset !== null) {
+                    $out .= '; name*=' . $this->_charset . '\'\'' . rawurlencode($this->_name);
+                }
             }
             if ($this->_filename) {
-                $out .= '; filename="' . $this->_filename . '"';
+                $out .= '; filename="' . preg_replace('/[^\x20-\x7E]/', '', $this->_filename) . '"';
```

Comment:
`name` and `filename` blocks looks same, add separate method to deduplicate.

It could also check if string contains any non-ASCII, and only then add `*=` value so in most cases it would leave as it was with simple `=` value.